src/QsCompiler/Core/SymbolTable.fs

```diff
@@ -261,21 +343,25 @@ and Namespace private
     let mutable TypesDefinedInAllSourcesCache = null
     let mutable CallablesDefinedInAllSourcesCache = null
 
-    /// Given a non-nullable string, returns true if either a callable or a type with that name already exists
-    /// either in a source file, or in a referenced assembly.
-    let IsDefined arg = 
-        CallablesInReferences.ContainsKey arg || TypesInReferences.ContainsKey arg || 
-        Parts.Values.Any (fun partial -> partial.ContainsCallable arg || partial.ContainsType arg)
+    /// Returns true if the name is available for use in a new declaration.
+    let isNameAvailable name =
```

Comment:
Could we keep with upper-casing? I know the lower case first letter is more aligned with F#, but we use upper case letters in the places where the expectation is that the property/method will be used from within C#. Just having everything upper case makes it more consistent I think.

Reply:
[F# naming conventions](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#naming-conventions) say to use camelCase everywhere except members and type names.

In terms of consistency, we already use lower case names in the Q# compiler for local let-bindings inside of functions, and let-bindings at the type level are basically local bindings inside the type constructor.

Reply:
As long as it's not visible, I am fine either way. But for let-bindings inside modules I'd like to go with upper case, since these may be exposed to C#, and it should not be visible for anyone using the compiler packages whether the code was writing in F# or C#.

Reply:
Yeah, surprisingly that page recommends even public module-level let-bindings be lowercased, but [on this page](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/component-design-guidelines#guidelines-for-f-facing-libraries) they recommend uppercase for identifiers that can be used from other .NET languages.

Should I rename all of my ResolutionResult module functions to start with an uppercase letter?

Reply:
Yes, I think it would be good to do that for the functions in the ResolutionResult. Thanks!